sljit_src/sljitExecAllocator.c

```diff
@@ -192,6 +192,63 @@ static SLJIT_INLINE void free_chunk(void *chunk, sljit_uw size)
 
 #endif
 
+#if defined(__APPLE__) && defined(__arm64__)
+__thread jit_protect_mode arm_current_jit_protect_mode = JPM_ENABLED;
+
+inline void sljit_arm_jit_write_protect_enable()
+{
+	if (__builtin_available(macOS 11, *)) {
+		if (arm_current_jit_protect_mode != JPM_ENABLED) {
+			pthread_jit_write_protect_np(1);
+			arm_current_jit_protect_mode = JPM_ENABLED;
+		}
+	}
+}
+
+inline void sljit_arm_jit_write_protect_disable()
+{
+	if (__builtin_available(macOS 11, *)) {
```

Comment:
this is most likely breaking when using gcc (or other compilers) instead of XCode/clang

Reply:
This code is guarded for macOS only and gcc is not even sure they will support compiling for Apple Silicon.

See gcc comment here for reference: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=96168#c6

Reply:
correct, but my point is that we should ideally set the expectations correctly and error out earlier if something other than XCode/clang is used to try to build this in macOS (ex: #error with a clear message) instead of hitting this line and throwing some sort of weird syntax error.

additionally; if we are giving up on using gcc a similar construct could be used to the code above that does MAP_JIT detection and which would make sense adding as a follow up commit.